File: katt/Contest3/iterated/include/python2/stub.py
import stub
import iterated
import sys
import os

N = -1
def _grader_init():
    global N
    global inf
    global outf
    inf = os.fdopen(os.open(sys.argv[1], os.O_RDONLY))
    outf = open(sys.argv[2], "w+")
    N = int(inf.readline())
    iterated.init(N)

def guess(ans):
    assert len(ans) == N
    assert all(x == True or x == False for x in ans)
    s = ''.join('1' if x else '0' for x in ans)
    outf.write(s)
    outf.write("\n")
    outf.flush()
    res = int(inf.readline())
    if res == N:
        sys.exit(0)
    return res

if __name__ != "__main__":
    stub._grader_init()

File: katt/Contest3/iterated/include/cpp/stub.cpp
#include "iterated.h"
#include <bits/stdc++.h>
using namespace std;

ifstream fin;
ofstream fout;

static int N;
int guess(const int ans[]) {
	string guess;
	guess.reserve(N);
	for (int i = 0; i < N; i++)
		guess += ans[i] ? '1' : '0';
	fout << guess << endl;
	int res;
	fin >> res;
	if (res == N)
		exit(0);
	return res;
}

int main(int argc, char** argv) {
  fin.open(argv[1], ifstream::in|ifstream::out);
  fout.open(argv[2], ofstream::out|ofstream::in);
  fin >> N;
	init(N);
}
